<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_5">
    <item>
     <widget class="QStackedWidget" name="stackedWidget">
      <property name="minimumSize">
       <size>
        <width>0</width>
        <height>40</height>
       </size>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="deployPage">
       <layout class="QVBoxLayout" name="verticalLayout_9">
        <item>
         <spacer name="verticalSpacer_9">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Policy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QWidget" name="deployList" native="true">
          <layout class="QVBoxLayout" name="verticalLayout_8">
           <item>
            <widget class="QLabel" name="deployLabel">
             <property name="text">
              <string>Deploy source and destinations</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="deployButtons">
             <item>
              <widget class="QPushButton" name="deployAddButton">
               <property name="text">
                <string>Add element</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="deployAddGroupButton">
               <property name="text">
                <string>Add group</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="deployClearButton">
               <property name="text">
                <string>Clear</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="deployRunAllButton">
               <property name="text">
                <string>Run all</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_6">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QScrollArea" name="deployScrollArea">
             <property name="widgetResizable">
              <bool>true</bool>
             </property>
             <widget class="QWidget" name="deployScrollAreaContents">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>774</width>
                <height>359</height>
               </rect>
              </property>
              <property name="sizePolicy">
               <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_7"/>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_8">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Policy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="deployFooterHLayout">
          <item>
           <widget class="QPushButton" name="settingsButton">
            <property name="text">
             <string>Settings</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer_8">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Expanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="saveButton">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="loadButton">
            <property name="text">
             <string>Load</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="settingsPage">
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Policy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="apiHLayout">
          <item>
           <widget class="QLabel" name="apiLabel">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>35</height>
             </size>
            </property>
            <property name="text">
             <string>API</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="apiLineEdit">
            <property name="placeholderText">
             <string>https://gitlab.example.com/api/v4</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="apiButtonUpdate">
            <property name="maximumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="text">
             <string>ðŸ—˜</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="userTokenHLayout">
          <property name="sizeConstraint">
           <enum>QLayout::SizeConstraint::SetDefaultConstraint</enum>
          </property>
          <item>
           <widget class="QLabel" name="userTokenLabel">
            <property name="minimumSize">
             <size>
              <width>120</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>16777215</width>
              <height>35</height>
             </size>
            </property>
            <property name="text">
             <string>User token</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="userTokenLineEdit">
            <property name="inputMask">
             <string/>
            </property>
            <property name="text">
             <string/>
            </property>
            <property name="placeholderText">
             <string>(optional if repo tokens are available)</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="userTokenButtonUpdate">
            <property name="maximumSize">
             <size>
              <width>35</width>
              <height>35</height>
             </size>
            </property>
            <property name="text">
             <string>ðŸ—˜</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QWidget" name="tokenList" native="true">
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <widget class="QLabel" name="tokenLabel">
             <property name="focusPolicy">
              <enum>Qt::FocusPolicy::StrongFocus</enum>
             </property>
             <property name="text">
              <string>Repository tokens</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="tokenButtons">
             <item>
              <widget class="QPushButton" name="tokenAddButton">
               <property name="text">
                <string>Add</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="tokenClearButton">
               <property name="text">
                <string>Clear</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_3">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QScrollArea" name="tokenScrollArea">
             <property name="widgetResizable">
              <bool>true</bool>
             </property>
             <widget class="QWidget" name="tokenScrollAreaContents">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>82</width>
                <height>16</height>
               </rect>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_11"/>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="sshList" native="true">
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <widget class="QLabel" name="sshLabel">
             <property name="focusPolicy">
              <enum>Qt::FocusPolicy::StrongFocus</enum>
             </property>
             <property name="text">
              <string>SSH access info</string>
             </property>
            </widget>
           </item>
           <item>
            <layout class="QHBoxLayout" name="sshButtons">
             <item>
              <widget class="QPushButton" name="sshAddButton">
               <property name="text">
                <string>Add</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="sshClearButton">
               <property name="text">
                <string>Clear</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer_2">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>40</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
            </layout>
           </item>
           <item>
            <widget class="QScrollArea" name="sshScrollArea">
             <property name="widgetResizable">
              <bool>true</bool>
             </property>
             <widget class="QWidget" name="sshScrollAreaContents">
              <property name="geometry">
               <rect>
                <x>0</x>
                <y>0</y>
                <width>82</width>
                <height>16</height>
               </rect>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_10"/>
             </widget>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer_4">
          <property name="orientation">
           <enum>Qt::Orientation::Vertical</enum>
          </property>
          <property name="sizeType">
           <enum>QSizePolicy::Policy::Preferred</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QHBoxLayout" name="settingsFooterHLayout">
          <item>
           <widget class="QPushButton" name="settingsBackButton">
            <property name="text">
             <string> &lt; Back </string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="horizontalSpacer">
            <property name="orientation">
             <enum>Qt::Orientation::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Policy::Expanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>40</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar">
   <property name="sizeGripEnabled">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <tabstops>
  <tabstop>apiLineEdit</tabstop>
  <tabstop>tokenLabel</tabstop>
  <tabstop>tokenAddButton</tabstop>
  <tabstop>tokenClearButton</tabstop>
  <tabstop>userTokenLineEdit</tabstop>
  <tabstop>sshLabel</tabstop>
  <tabstop>sshAddButton</tabstop>
  <tabstop>sshClearButton</tabstop>
  <tabstop>deployAddButton</tabstop>
  <tabstop>deployClearButton</tabstop>
  <tabstop>deployRunAllButton</tabstop>
  <tabstop>deployScrollArea</tabstop>
  <tabstop>saveButton</tabstop>
  <tabstop>loadButton</tabstop>
  <tabstop>settingsButton</tabstop>
  <tabstop>sshScrollArea</tabstop>
  <tabstop>tokenScrollArea</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
